packages/ra-core/src/controller/list/useSelectAll.tsx

```diff
@@ -0,0 +1,117 @@
+import { useCallback } from 'react';
+import { useQueryClient } from '@tanstack/react-query';
+
+import { useNotify } from '../../notification';
+import { useDataProvider, UseGetListOptions } from '../../dataProvider';
+import type { FilterPayload, RaRecord, SortPayload } from '../../types';
+import type { UseRecordSelectionResult } from './useRecordSelection';
+
+/**
+ * Function hook to select all items of a list (until we reached the limit)
+ *
+ * @param {string} resource Required. The resource name
+ * @param {RecordSelectionModifiers} selectionModifiers Required, The selection modifiers
+ * @param {SortPayload} sort Optional. The sort object passed to the dataProvider
+ * @param {FilterPayload} filter Optional. The filter object passed to the dataProvider
+ * @returns {Function} onSelectAll A function to select all items of a list
+ *
+ * @example
+ *
+ * const MyButton = () => {
+ *   const [_, selectionModifiers] = useRecordSelection({ resource: 'posts' });
```

Comment:
This example doesn't make sense: the record selection will be local to this button and can't be used to perform any action.